modules/azure/servicebus.go

```diff
@@ -0,0 +1,383 @@
+package azure
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/servicebus/mgmt/2017-04-01/servicebus"
+	"github.com/stretchr/testify/require"
+)
+
+func serviceBusNamespaceClientE(subscriptionID string) (*servicebus.NamespacesClient, error) {
+	authorizer, err := NewAuthorizer()
+	if err != nil {
+		return nil, err
+	}
+
+	nsClient := servicebus.NewNamespacesClient(subscriptionID)
+	nsClient.Authorizer = *authorizer
+	return &nsClient, nil
+}
+
+func serviceBusTopicClientE(subscriptionID string) (*servicebus.TopicsClient, error) {
+	authorizer, err := NewAuthorizer()
+	if err != nil {
+		return nil, err
+	}
+
+	tClient := servicebus.NewTopicsClient(subscriptionID)
+	tClient.Authorizer = *authorizer
+	return &tClient, nil
+}
+
+func serviceBusSubscriptionsClientE(subscriptionID string) (*servicebus.SubscriptionsClient, error) {
+	authorizer, err := NewAuthorizer()
+	if err != nil {
+		return nil, err
+	}
+
+	sClient := servicebus.NewSubscriptionsClient(subscriptionID)
+	sClient.Authorizer = *authorizer
+	return &sClient, nil
+}
+
+// ListServiceBusNamespaceE list all SB namespaces in all resource groups in the given subscription ID. This function would fail the test if there is an error.
+func ListServiceBusNamespaceE(subscriptionID string) (*[]servicebus.SBNamespace, error) {
+	nsClient, err := serviceBusNamespaceClientE(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	iteratorSBNamespace, err := nsClient.ListComplete(context.Background())
+	if err != nil {
+		return nil, err
+	}
+
+	results := make([]servicebus.SBNamespace, 0)
+	for iteratorSBNamespace.NotDone() {
+		results = append(results, iteratorSBNamespace.Value())
+		err = iteratorSBNamespace.Next()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &results, nil
+}
+
+// ListServiceBusNamespace - list all SB namespaces in all resource groups in the given subscription ID.
+func ListServiceBusNamespace(t *testing.T, subscriptionID string) *[]servicebus.SBNamespace {
+	results, err := ListServiceBusNamespaceE(subscriptionID)
+	
+	require.NoError(t, err)
+
+	return results
+}
+
+// ListServiceBusNamespaceNamesE list names of all SB namespaces in all resource groups in the given subscription ID. This function would fail the test if there is an error.
+func ListServiceBusNamespaceNamesE(subscriptionID string) (*[]string, error) {
+	sbNamespace, err := ListServiceBusNamespaceE(subscriptionID)
+
+	if err != nil {
+		return nil, err
+	}
+
+	results := make([]string, 0)
+	for _, namespace := range *sbNamespace {
+		results = append(results, *namespace.Name)
```

Comment:
Use [safePtrToString](https://github.com/gruntwork-io/terratest/blob/master/modules/azure/common.go#L56) when dereferencing string ptr attributes.

Reply:
@HadwaAbdelhalem looks like this comment wasn't addressed. I'm assuming that's because there is some guarantee in the SDK that ensures this is set. Is that correct assumption?

Reply:
@yorinasub17 it is, the SDK is calling the Azure Rest API underneath for all resources operations, and the validation of the the name as a required parameter is done on the rest API [Here](https://docs.microsoft.com/en-us/rest/api/servicebus/stable/namespaces/create-or-update)